cms/private_media/tests/test_serve_views.py

```diff
@@ -0,0 +1,150 @@
+from unittest import mock
+
+from django.contrib.auth import get_user_model
+from django.test import TestCase, override_settings
+from wagtail.documents import get_document_model
+from wagtail.images import get_image_model
+from wagtail.images.models import Filter
+from wagtail.models import Collection
+from wagtail_factories import DocumentFactory, ImageFactory
+
+from cms.private_media.constants import Privacy
+
+
+class TestImageServeView(TestCase):
+    model = get_image_model()
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.root_collection = Collection.objects.get(depth=1)
+        cls.private_image = ImageFactory(collection=cls.root_collection)
+        cls.public_image = ImageFactory(_privacy=Privacy.PUBLIC, collection=cls.root_collection)
+        cls.private_image_renditions = cls.private_image.create_renditions(Filter("fill-10x10"), Filter("fill-20x20"))
+        cls.public_image_renditions = cls.public_image.create_renditions(Filter("fill-10x10"), Filter("fill-20x20"))
+        cls.superuser = get_user_model().objects.create(username="superuser", is_superuser=True)
+
+    def test_serve_private_image(self):
+        """Test the serve view behaviour for private image renditions."""
+        # If not authenticated, permission checks should fail and a Forbidden response returned
+        for rendition in self.private_image_renditions.values():
+            for is_external_env in [True, False]:
+                with self.subTest(is_external_env=is_external_env) and override_settings(
+                    IS_EXTERNAL_ENV=is_external_env
+                ):
+                    response = self.client.get(rendition.serve_url)
+                    self.assertEqual(response.status_code, 403)
+
+        # If authenticated as a superuser, the view should serve the files
+        self.client.force_login(self.superuser)
+        for rendition in self.private_image_renditions.values():
+            for is_external_env in [True, False]:
+                with self.subTest(is_external_env=is_external_env) and override_settings(
+                    IS_EXTERNAL_ENV=is_external_env
+                ):
+                    response = self.client.get(rendition.serve_url)
+                    self.assertEqual(response.status_code, 200)
+
+    def test_serve_public_image(self):
+        """Test the serve view behaviour for public image renditions."""
+        # For public image renditions, the serve view should redirect to the file path orURL.
```

Comment:
```suggestion
        # For public image renditions, the serve view should redirect to the file path or URL.
```

Reply:
Resolved in [1dcdcb3](https://github.com/ONSdigital/dis-wagtail/pull/89/commits/1dcdcb311e437d3b5043cd407bf131536a25e423)